### PR TITLE
Add testing lib jest dom types, fixes #311

### LINKED ISSUE #311: [cli] a newly scaffolded app should include type definitions for react testing library
Trying to use expect methods from react testing library will generate a typescript error, for example

```typescript
expect(rendered.baseElement).toBeInTheDocument()
```

will result in `Property 'toBeInTheDocument' does not exist on type 'JestMatchersShape<Matchers<void, HTMLElement>, Matchers<Promise<void>, HTMLElement>>'.`

### PR DESCRIPTION
**Pull request by nikek**
#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [x] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] [CODEOWNERS](./CODEOWNERS) updated (for new stuff)
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs [hunk 1]
```diff
@@ -17,10 +17,15 @@
...
+import { ThemeProvider } from '@material-ui/core';
+import { BackstageTheme } from '@spotify-backstage/core';
 
 describe('ExampleComponent', () => {
   it('should render', () => {
-    const rendered = render(<ExampleComponent />);
+    const rendered = render(
+      <ThemeProvider theme={BackstageTheme}>
+        <ExampleComponent />
+      </ThemeProvider>,);
```
Comment (freben) [inline packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs]: Does this lint?

**Review by freben — APPROVED** [review]

File: packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs (hunk 1 above)
Comment (nikek) [reply packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs]: I'll remove the comma

Comment (freben) [reply packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs]: If you set vscode to tax (in the lower right corner) it'll lint/prettier it for you

Comment (nikek) [reply packages/cli/templates/default-plugin/src/components/ExampleComponent/ExampleComponent.test.tsx.hbs]: Oh, wait, prettier is adding the comma. Also, regarding linting, I think there's a bunch of things that isn't linting right now, we need to go through those warnings and clean up.

**Pull request merged**